cmd/fluxctl/diff_cmd.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/weaveworks/flux/diff"
+	kdiff "github.com/weaveworks/flux/platform/kubernetes/diff"
+)
+
+type diffOpts struct {
+	*rootOpts
+	format string
+}
+
+func newDiff(root *rootOpts) *diffOpts {
+	return &diffOpts{rootOpts: root}
+}
+
+func (opts *diffOpts) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "diff",
+		Short: "Show differences between one platform config and another",
+		RunE:  opts.RunE,
+	}
+	cmd.Flags().StringVarP(&opts.format, "output", "o", "text", "(yaml|json|text) whether to output differences in YAML or JSON, or just summarise in text")
+	return cmd
+}
+
+func (opts *diffOpts) RunE(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return newUsageError("please supply two filenames")
+	}
+
+	output := func(diffs diff.ObjectSetDiff) error {
+		diffs.Summarise(cmd.OutOrStdout())
+		return nil
+	}
+
+	switch opts.format {
```

Comment:
This is a bit pointless; I mainly wanted it for seeing the raw (`printf(%s)`) output for debugging.